server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/routes/HealthCheckRoutes.java

```diff
@@ -77,15 +81,25 @@ public void define(Service service) {
             @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
                 message = "Internal server error - When one check has failed.")
     })
-    public void validateHealthchecks() {
-        service.get(HEALTHCHECK,
-            (request, response) -> {
-                List<Result> anyUnhealthyOrDegraded = retrieveUnhealthyOrDegradedHealthChecks();
-
-                anyUnhealthyOrDegraded.forEach(this::logFailedCheck);
-                response.status(getCorrespondingStatusCode(anyUnhealthyOrDegraded));
-                return response;
-            });
+	public Object validateHealthchecks(Request request, Response response) {
+		List<Result> anyUnhealthyOrDegraded = retrieveUnhealthyOrDegradedHealthChecks();
+		anyUnhealthyOrDegraded.forEach(this::logFailedCheck);
+		response.status(getCorrespondingStatusCode(anyUnhealthyOrDegraded));
+		return response;
+	}
+    
+    @GET
+    @Path(CHECKS)
+    @ApiOperation(value = "List all health checks")
+    @ApiResponses(value = {
+            @ApiResponse(code = HttpStatus.OK_200, message = "List of all health checks", 
+            		response = HealthCheckDto.class, responseContainer = "List"),
+            @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "No health checks found")
```

Comment:
I think if no healthcheck is found we just return empty, no?

Reply:
I thought of returning a 404 if the checks are empty. Will remove that if you suggest.